pkg/protocol/engine/attestation/slotattestation/manager.go

```diff
@@ -319,6 +320,31 @@ func (m *Manager) Rollback(targetSlot iotago.SlotIndex) error {
 	return nil
 }
 
+// Reset resets the component to a clean state as if it was created at the last commitment.
+func (m *Manager) Reset() {
+	futureAttestationsToClear := make([]iotago.SlotIndex, 0)
+	m.futureAttestations.ForEach(func(slot iotago.SlotIndex, _ *shrinkingmap.ShrinkingMap[iotago.AccountID, *iotago.Attestation]) {
+		if slot > m.lastCommittedSlot {
+			futureAttestationsToClear = append(futureAttestationsToClear, slot)
+		}
+	})
+
+	pendingAttestationsToClear := make([]iotago.SlotIndex, 0)
+	m.pendingAttestations.ForEach(func(slot iotago.SlotIndex, _ *shrinkingmap.ShrinkingMap[iotago.AccountID, *iotago.Attestation]) {
```

Comment:
Pending attestations contain only attestations that only belong to slots that were committed, so actually it might require a different reset strategy. I'm not sure about that so it's better to check with Jonas.

Reply:
I talked to Jonas and we dont need to clean the pending ones at all.